syncer/db.go

```diff
@@ -66,208 +68,87 @@ type binlogSize struct {
 
 // Conn represents a live DB connection
 type Conn struct {
-	cfg *config.SubTaskConfig
-
-	db *sql.DB
+	cfg      *config.SubTaskConfig
+	baseConn *utils.BaseConn
 }
 
-func (conn *Conn) querySQL(tctx *tcontext.Context, query string, maxRetry int) (*sql.Rows, error) {
-	if conn == nil || conn.db == nil {
+func (conn *Conn) querySQL(tctx *tcontext.Context, query string) (*sql.Rows, error) {
+	if conn == nil || conn.baseConn == nil {
 		return nil, errors.NotValidf("database connection")
 	}
 
-	var (
-		err  error
-		rows *sql.Rows
-	)
-	tctx.L().Debug("query statement", zap.String("sql", query))
-	for i := 0; i < maxRetry; i++ {
-		if i > 0 {
-			sqlRetriesTotal.WithLabelValues("query", conn.cfg.Name).Add(1)
-			tctx.L().Warn("query statement", zap.Int("retry", i), zap.String("sql", query))
-			time.Sleep(retryTimeout)
-		}
-
-		rows, err = conn.db.QueryContext(tctx.Context(), query)
-		if err != nil {
-			if !isRetryableError(err) {
-				return rows, errors.Trace(err)
+	ret, err := conn.baseConn.NormalRetryOperation(
+		tctx,
+		func(ctx *tcontext.Context, _ int) (interface{}, error) {
+			rows, err := conn.baseConn.QuerySQL(ctx, query)
+			return rows, err
+		},
+		func(retryTime int, err error) bool {
+			if isRetryableError(err) {
+				sqlRetriesTotal.WithLabelValues("query", conn.cfg.Name).Add(1)
+				tctx.L().Warn("query statement failed and retry", zap.Int("retry", retryTime), zap.String("sql", query), log.ShortError(err))
+				return true
 			}
-			tctx.L().Warn("query statement failed and retry", zap.String("sql", query), log.ShortError(err))
-			continue
-		}
-
-		return rows, nil
-	}
+			return false
+		})
 
 	if err != nil {
 		tctx.L().Error("query statement failed", zap.String("sql", query), log.ShortError(err))
 		return nil, errors.Trace(err)
 	}
-
-	return nil, errors.Errorf("query sql[%s] failed", query)
+	return ret.(*sql.Rows), nil
 }
 
-// Note: keep it for later use?
-func (conn *Conn) executeSQL(tctx *tcontext.Context, sqls []string, args [][]interface{}, maxRetry int) error {
-	if len(sqls) == 0 {
-		return nil
+func (conn *Conn) executeSQL(tctx *tcontext.Context, queries []string, args [][]interface{}) (int, error) {
```

Comment:
some function use sqls, some use queries, I think it is better to uniform them

Reply:
`sqls` and `queries` are different
`sqls` is an array of `query with argument`
`queries` is an array of `query without argument`